controllers/machinehealthcheck_controller.go

```diff
@@ -632,5 +635,38 @@ func (r *MachineHealthCheckReconciler) externalRemediationRequestExists(ctx cont
 	if err != nil {
 		return false
 	}
+	logger := ctrl.LoggerFrom(ctx)
+	if r.isAlertOldEmr(remediationReq, logger) {
+		//TODO mshitrit send alert here
+		//metrics.ObserveMachineHealthCheckOldEmr(m.Name, m.Namespace)
+	}
 	return remediationReq != nil
 }
+
+func (r *MachineHealthCheckReconciler) isAlertOldEmr(emr *unstructured.Unstructured, logger logr.Logger) bool {
+	//verify emr exist
+	if emr == nil {
+		return false
+	}
+	isSendAlert := false
+	//verify emr is old
+	if time.Now().After(emr.GetCreationTimestamp().Add(oldEmrAlertTimeout)) {
+		var emrAnnotations map[string]string
+		if emrAnnotations = emr.GetAnnotations(); emrAnnotations == nil {
+			emrAnnotations = map[string]string{}
+		}
+		//verify this is the first alert for this emr
+		if _, isAlertedSent := emrAnnotations[oldEmrAnnotationKey]; !isAlertedSent {
+			emrAnnotations[oldEmrAnnotationKey] = "flagon"
+			emr.SetAnnotations(emrAnnotations)
+			if err := r.Client.Update(context.TODO(), emr); err == nil {
```

Comment:
still trying to understand what we are trying to achieve.
This new annotation applies to the EMR CR, which is reconciled by the EMR controller.
If this is the case, why CAPI should take care of creating the annotation? IMO it make a lot more sense that each EMR controller determine when to consider its own CR old (which can vary implementation by implementation) and to take appropriate actions

Reply:
Uhmm, I don't think we should assign to CAPI the responsibility to provide a signal that other controllers are not working. 

In CAPI we using `status.ObservedGeneration` as a way to convey an evidence that the controllers are properly working, and other controllers are doing the same outside CAPI as well (if I'm not wrong in main Kubernetes as well).
 
So, if the goal of this change is to provide evidence that EMR controllers are not working, I would recommend that EMR controllers apply the same pattern by implementing `status.ObservedGeneration` instead of relying on CAPI for doing so.

Reply:
Agree with what @fabriziopandini said above. We should also have had an issue discussing this, or more details in the PR itself.

Reply:
@elmiko Could we open an issue about it? I'm still not sure if Cluster API should be in charge of alerting if other controllers timeout, usually progress gets stuck in a specific step

Reply:
@fabriziopandini  I'm fine with closing the PR.
@vincepri , @elmiko I just want to make sure it's OK by you as well, before I do it.

Reply:
no objection from me to close this. i still need to make an issue documenting the underlying concern i have, but that shouldn't affect this. thanks @mshitrit !